From 913d55037616659c04763e756f948fcbaef0bbee Mon Sep 17 00:00:00 2001
Message-Id: <913d55037616659c04763e756f948fcbaef0bbee.1648734715.git.geliang.tang@suse.com>
From: He Fengqing <hefengqing@huawei.com>
Date: Tue, 30 Mar 2021 02:48:43 +0000
Subject: [PATCH] bpf: Remove unused bpf_load_pointer
Git-commit: 913d55037616659c04763e756f948fcbaef0bbee
Patch-mainline: v5.13-rc1
References: bsn#127
Modified-by-SEL: No


Remove unused bpf_load_pointer function in filter.h. The last user of it has
been removed with 24dea04767e6 ("bpf, x32: remove ld_abs/ld_ind").

Signed-off-by: He Fengqing <hefengqing@huawei.com>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
Acked-by: Song Liu <songliubraving@fb.com>
Link: https://lore.kernel.org/bpf/20210330024843.3479844-1-hefengqing@huawei.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 include/linux/filter.h | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/include/linux/filter.h b/include/linux/filter.h
index eecfd82db648..9a09547bc7ba 100644
--- a/include/linux/filter.h
+++ b/include/linux/filter.h
@@ -1246,15 +1246,6 @@ static inline u16 bpf_anc_helper(const struct sock_filter *ftest)
 void *bpf_internal_load_pointer_neg_helper(const struct sk_buff *skb,
 					   int k, unsigned int size);
 
-static inline void *bpf_load_pointer(const struct sk_buff *skb, int k,
-				     unsigned int size, void *buffer)
-{
-	if (k >= 0)
-		return skb_header_pointer(skb, k, size, buffer);
-
-	return bpf_internal_load_pointer_neg_helper(skb, k, size);
-}
-
 static inline int bpf_tell_extensions(void)
 {
 	return SKF_AD_MAX;
-- 
2.34.1

